src/main/java/com/minecolonies/commands/AbstractSplitCommand.java

```diff
@@ -0,0 +1,118 @@
+package com.minecolonies.commands;
+
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A command that has children.
+ */
+public abstract class AbstractSplitCommand implements ISubCommand
+{
+
+    private final String[] parents;
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public AbstractSplitCommand(@NotNull String... parents)
+    {
+        this.parents = parents;
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        final Map<String, ISubCommand> childs = getSubCommands();
+        final StringBuilder sb = new StringBuilder().append('/');
+        for (final String parent : parents)
+        {
+            sb.append(parent).append(' ');
+        }
+        sb.append('<');
+        boolean first = true;
+        for (final String child : childs.keySet())
+        {
+            if (first)
+            {
+                first = false;
+            }
+            else
+            {
+                sb.append('|');
+            }
+            sb.append(child);
+        }
+        sb.append('>');
+        return sb.toString();
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String[] args) throws CommandException
+    {
+        final Map<String, ISubCommand> childs = getSubCommands();
+        if (args.length == 0
+              || !childs.containsKey(args[0]))
+        {
+            //todo: check if WrongUsageException is better
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1135)